File: PYTHON/Tiket.py
class Tiket:

    # constructor
    def __init__(self, kodeFilm:str, namaFilm:str, harga:int, nomorKursi:str, namaBioskop:str):
        self.__kodeFilm = str(kodeFilm)
        self.__namaFilm = str(namaFilm)
        self.__harga = int(harga)
        self.__nomorKursi = str(nomorKursi)
        self.__namaBioskop = str(namaBioskop)

    # getter
    def getkodeFilm(self) -> str:
        return self.__kodeFilm

    def getnamaFilm(self) -> str:
        return self.__namaFilm

    def getharga(self) -> int:
        return self.__harga

    def getnomorKursi(self) -> str:
        return self.__nomorKursi

    def getnamaBioskop(self) -> str:
        return self.__namaBioskop

    # setter
    def setkodeFilm(self, kodeFilm:str) -> None:
        self.__kodeFilm = str(kodeFilm)

    def setnamaFilm(self, namaFilm:str) -> None:
        self.__namaFilm = str(namaFilm)

    def setharga(self, harga:int) -> None:
        self.__harga = int(harga)

    def setnomorKursi(self, nomorKursi:str) -> None:
        self.__nomorKursi = str(nomorKursi)

    def setnamaBioskop(self, namaBioskop:str) -> None:
        self.__namaBioskop = str(namaBioskop)

File: PHP/Tiket.php
<?php

class Tiket {
    private string $kodeTiket;
    private string $namaFilm;
    private int $harga;
    private string $nomorKursi;
    private string $namaBioskop;
    private string $gambar; // path file lokal gambar tiket/poster film

    public function __construct(
        string $kodeTiket = "",
        string $namaFilm = "",
        int $harga = 0,
        string $nomorKursi = "",
        string $namaBioskop = "",
        string $gambar = ""
    ) {
        $this->kodeTiket = $kodeTiket;
        $this->namaFilm = $namaFilm;
        $this->harga = $harga;
        $this->nomorKursi = $nomorKursi;
        $this->namaBioskop = $namaBioskop;
        $this->gambar = $gambar;
    }

    // getter
    public function getKodeTiket(): string { return $this->kodeTiket; }
    public function getNamaFilm(): string { return $this->namaFilm; }
    public function getHarga(): int { return $this->harga; }
    public function getNomorKursi(): string { return $this->nomorKursi; }
    public function getNamaBioskop(): string { return $this->namaBioskop; }
    public function getGambar(): string { return $this->gambar; }

    // setter
    public function setKodeTiket(string $kodeTiket): void { $this->kodeTiket = $kodeTiket; }
    public function setNamaFilm(string $namaFilm): void { $this->namaFilm = $namaFilm; }
    public function setHarga(int $harga): void { $this->harga = $harga; }
    public function setNomorKursi(string $nomorKursi): void { $this->nomorKursi = $nomorKursi; }
    public function setNamaBioskop(string $namaBioskop): void { $this->namaBioskop = $namaBioskop; }
    public function setGambar(string $gambar): void { $this->gambar = $gambar; }

    // ubah object jadi array asosiatif, supaya gampang disimpan di $_SESSION
    public function toArray(): array {
        return [
            'kodeTiket'   => $this->kodeTiket,
            'namaFilm'    => $this->namaFilm,
            'harga'       => $this->harga,
            'nomorKursi'  => $this->nomorKursi,
            'namaBioskop' => $this->namaBioskop,
            'gambar'      => $this->gambar,
        ];
    }

    // bikin object Tiket dari array asosiatif (kebalikan dari toArray)
    public static function fromArray(array $data): Tiket {
        return new Tiket(
            $data['kodeTiket'] ?? "",
            $data['namaFilm'] ?? "",
            (int)($data['harga'] ?? 0),
            $data['nomorKursi'] ?? "",
            $data['namaBioskop'] ?? "",
            $data['gambar'] ?? ""
        );
    }
}
